src/Shared/Metrics/DefaultMeterFactory.cs

```diff
@@ -0,0 +1,108 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.Metrics;
+using System.Linq;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.Metrics;
+
+internal sealed class DefaultMeterFactory : IMeterFactory
+{
+    private readonly IOptions<MetricsOptions> _options;
+    private readonly IMeterRegistry _meterRegistry;
+    private readonly Dictionary<MeterKey, Meter> _meters = new Dictionary<MeterKey, Meter>();
+
+    public DefaultMeterFactory(IOptions<MetricsOptions> options, IMeterRegistry meterRegistry)
+    {
+        _options = options;
+        _meterRegistry = meterRegistry;
+    }
+
+    public Meter CreateMeter(string name)
+    {
+        return CreateMeterCore(name, version: null, defaultTags: null);
+    }
+
+    public Meter CreateMeter(MeterOptions options)
+    {
+        return CreateMeterCore(options.Name, options.Version, options.DefaultTags);
+    }
+
+    private Meter CreateMeterCore(string name, string? version, IList<KeyValuePair<string, object?>>? defaultTags)
+    {
+        var tags = defaultTags?.ToArray();
+        if (tags != null)
+        {
+            Array.Sort(tags, (t1, t2) => string.Compare(t1.Key, t2.Key, StringComparison.Ordinal));
+        }
+        var key = new MeterKey(name, version, tags);
+
+        if (_meters.TryGetValue(key, out var meter))
+        {
+            return meter;
+        }
+
+        // TODO: Configure meter with default tags.
```

Comment:
Is this still TODO?

Reply:
It's still TODO. Meter currently doesn't support default tags. That change needs to happen in System.Diagnostics.Metrics.